src/server-extension/resolver.ts

```diff
@@ -0,0 +1,58 @@
+import "reflect-metadata";
+import { GraphQLResolveInfo } from "graphql";
+import { Arg, Info, Query, Resolver } from "type-graphql";
+import type { EntityManager } from "typeorm";
+import { ProjectType } from "./types"; // Custom ProjectType
+import { getSelectedFields } from "./helper";
+
+@Resolver()
+export class ProjectResolver {
+  constructor(private tx: () => Promise<EntityManager>) {}
+
+  @Query(() => [ProjectType])
+  async getProjectsSortedByVouchOrFlag(
+    @Arg("orgIds", () => [String]) orgIds: string[],
+    @Arg("sortBy", () => String) sortBy: "vouch" | "flag",
+    @Arg("limit", () => Number, { nullable: true }) limit: number = 10,
+    @Arg("offset", () => Number, { nullable: true }) offset: number = 0,
+    @Info() info: GraphQLResolveInfo
+  ): Promise<ProjectType[]> {
+    try {
+      const manager = await this.tx();
+
+      const vouchValue = sortBy === "vouch" ? true : false;
+
+      const selectedFields = getSelectedFields(info);
+      const fields = selectedFields.join(", ");
+
+      const query = `
+        SELECT 
+          ${fields}, 
```

Comment:
Note, you can only select fields either mentioned in the group by section or aggregated version of them